lib/Sema/TypeCheckRequestFunctions.cpp

```diff
@@ -73,13 +73,29 @@ Type InheritedTypeRequest::evaluate(
 
   const TypeLoc &typeLoc = InheritedTypes(decl).getEntry(index);
 
+  if (auto typeRepr = typeLoc.getTypeRepr()) {
+    if (auto itr = dyn_cast<InverseTypeRepr>(typeRepr)) {
+      Type inheritedTy = resolution->resolveType(itr->getConstraint());
+      auto kp = inheritedTy->getKnownProtocol();
+      if (kp == KnownProtocolKind::BitwiseCopyable) {
```

Comment:
Can this be generalized to handle any known protocol (that isn't invertible, I guess), so there are fewer places to update when we want to thread through one of the others?

Reply:
Changed to return any non-invertible known protocol and the TypeRepr (for diagnostics).